main.js

```diff
@@ -1,3 +1,29 @@
 import './style.css';
 
-console.log('Hello world');
+const modalButton = document.getElementById('modal-button');
+const modal = document.getElementById('modal');
+const modalCloseButton = document.getElementById('modal-close-button');
+const toolTipButton = document.getElementById('tooltip-button');
+const toolTipText = document.getElementById('tooltip');
+
+modalButton.addEventListener('click', () => {
+    modal.style.display = 'block';
+});
+
+modalCloseButton.addEventListener('click', () => {
+    modal.style.display = 'none';
+});
+
+toolTipButton.addEventListener('click', () => {
+    if (toolTipText.style.visibility === 'visible') {
+        toolTipText.style.visibility = 'hidden'
+    } else{
+        toolTipText.style.visibility = 'visible'
+    }
+})
+
+window.addEventListener('click', (e) => {
```

Comment:
Why do we need to listen window?

Reply:
because I want to close modal window on tap to any region outside modal window

Reply:
Got it! It is one of ways to solve it
For your aim it might be enough to add event to document, or document.body.
But i found this article about Firefox issues with that:
https://gomakethings.com/you-should-always-attach-your-vanilla-js-click-events-to-the-window/

So, you can add the same behaviour like that:
```javascript
document.documentElement.addEventListener('click', ...)
``` 

With that you will add listener to the top level of document, not to the window, which is kinda out of DOM scope.

Anyway. All variants are fine for our education, but you can see how tricky it can be 🙂